cmd2/argparse_completer.py

```diff
@@ -704,32 +701,17 @@ def _resolve_choices_for_arg(self, action: argparse.Action, used_values=()) -> L
             # is the argument a string? If so, see if we can find an attribute in the
             # application matching the string.
             if isinstance(args, str):
-                try:
-                    args = getattr(self._cmd2_app, args)
-                except AttributeError:
-                    # Couldn't find anything matching the name
-                    return []
+                args = getattr(self._cmd2_app, args)
 
             # is the provided argument a callable. If so, call it
             if callable(args):
                 try:
-                    try:
-                        args = args(self._cmd2_app)
-                    except TypeError:
-                        args = args()
+                    args = args(self._cmd2_app)
                 except TypeError:
-                    return []
-
-            try:
-                iter(args)
```

Comment:
You dropped the `iter(args)` line which is what might have generated an exception prior to the next line.  I'm not sure what the effect would be, but are you sure this is the correct modification?

Reply:
Nevermind.  The list comprehension will throw an exception if args isn't iterable, so we don't need an explicit check.